src/validators.js

```diff
@@ -4,7 +4,7 @@ const validateEmail = (email) => {
   return new Promise((resolve, reject) => {
```

Comment:
this doesn't do anything asynchronous so you don't need to return a promise

Reply:
if _not_ valid, you could throw an error, that should mean you get similar behaviour as with a rejected promise